controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -172,7 +172,7 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 	}
 
 	// Override with MCG options specified in the storage cluster spec
-	if sc.Spec.MultiCloudGateway != nil {
+	if sc.Spec.MultiCloudGateway != nil && sc.Spec.MultiCloudGateway.DbStorageClassName != "" {
```

Comment:
Is this change needed for this feature to work?

Reply:
It is not related to the feature but to avoid the loop since it will not be nil if pgsecrets are not nil